### PR TITLE
add R16B compatibility

### PR DESCRIPTION
**Pull request by kuenishi**
in favor of https://github.com/basho/velvet/pull/9 . I'll push stanchion's PR later.

### CODE REVIEW

File: src/riak_cs_s3_auth.erl [hunk 1]
```diff
@@ -86,6 +86,9 @@ parse_auth_header("AWS " ++ Key) ->
 parse_auth_header(_) ->
     {undefined, undefined}.
 
+%% TODO:
+%% can this be replaced with stanchion_auth:request_sinature/2?
```
Comment (kellymclaughlin) [inline src/riak_cs_s3_auth.erl]: I would say no it cannot. Better to not call any stanchion modules directly from `riak_cs` code. If anything we should make functions used by both apps into libraries, but we are punting on most of that for now until there is more certainty about the future of stanchion.

File: src/riak_cs_s3_auth.erl [hunk 2]
```diff
@@ -127,8 +130,14 @@ calculate_signature(KeyData, RD) ->
            AmazonHeaders,
            Resource],
     _ = lager:debug("STS: ~p", [STS]),
-    base64:encode_to_string(
-      crypto:sha_mac(KeyData, STS)).
+
+    %% This is for compatibility for R16B and R15B
```
Comment (kellymclaughlin) [inline src/riak_cs_s3_auth.erl]: Yeah, they introduced these new function in R15B03 of all things. Ugh!

File: src/riak_cs_s3_auth.erl (hunk 1 above)
Comment (kuenishi) [inline src/riak_cs_s3_auth.erl]: Ah, I see. I'll stay tuned, looking forward to rafter.

File: src/riak_cs_s3_auth.erl (hunk 2 above)
Comment (kuenishi) [inline src/riak_cs_s3_auth.erl]: OK, I'll put same workaround as Riak did. Ugggghhhhhh

File: src/riak_cs_wm_users.erl [hunk 3]
```diff
@@ -180,6 +180,6 @@ user_fold_fun(RiakPid, Status) ->
     end.
 
 unique_id() ->
-    Rand = crypto:sha(term_to_binary({make_ref(), now()})),
+    Rand = riak_cs_utils:sha(term_to_binary({make_ref(), now()})),
```
Comment (shino) [inline src/riak_cs_wm_users.erl:183]: i got `undef`

Comment (kuenishi) [inline src/riak_cs_wm_users.erl:183]: fixed! could you please test it again?

**Comment by shino** [comment]
R15B01
- [x] Eunit and eqc tests pass
- [x] Pulse tests
- [x] Dialyzer output clean
- [x] `riak_test` tests pass (with https://github.com/basho/stanchion/pull/59 and https://github.com/basho/velvet/pull/9)

R16B01
- [x] Eunit and eqc tests pass
- [x] Pulse tests
- [x] Dialyzer output clean
- [x] `riak_test` tests pass (with https://github.com/basho/stanchion/pull/59 and https://github.com/basho/velvet/pull/9)

:+1:

**Pull request merged**